src/_common/settings/options.service.ts

```diff
@@ -0,0 +1,74 @@
+import { assertNever } from '../../utils/utils';
+const STORAGE_PREFIX = 'options.';
+
+type OptionsType = 'community-eject' | 'community-move-post' | 'community-user-block';
+
+export function getOptions(type: OptionsType, suffix = '') {
+	const id = type + (suffix ? '-' + suffix : '');
+	switch (type) {
+		case 'community-eject':
+		case 'community-move-post':
+			return new Options<string>(id, true, 10);
+		case 'community-user-block':
+			return new Options<string>(id, true, 20);
+		default:
+			assertNever(type);
+	}
+}
+
+export class Options<T> {
+	private _storageKey: string;
+	private _checkDuplicate: boolean;
+	private _maxItems: number | null;
+
+	constructor(id: string, checkDuplicate: boolean, maxItems: number | null) {
+		this._storageKey = STORAGE_PREFIX + id;
+		this._checkDuplicate = checkDuplicate;
+		this._maxItems = maxItems;
+	}
+
+	private _getStorageValue(): string {
+		const value = localStorage.getItem(this._storageKey);
+		if (value === null) {
+			return '[]';
+		}
+		return value;
+	}
+
+	private _setStorageValue(value: string) {
+		localStorage.setItem(this._storageKey, value);
+	}
+
+	public getList(): T[] {
+		const list = JSON.parse(this._getStorageValue());
+		if (!Array.isArray(list)) {
+			return [];
+		}
+		return list;
+	}
+
+	public setList(list: T[]) {
+		const value = JSON.stringify(list);
+		this._setStorageValue(value);
+	}
+
+	public unshiftItem(item: T) {
```

Comment:
I think it'd be nice to enforce some max serialized length on the item as a sanity check, just to avoid filling the localstorage. its a fairly limited resource and the data list autocomplete won't show long results in full anyways. lets try to keep this minimal and practical, esp since we might add actual templating support in the future yeah?

Reply:
Well, it is inherently limited in length due to the input fields it uses. Since we control when/where we use this, we should be good with this rn.